xwiki-platform-core/xwiki-platform-oldcore/src/main/java/com/xpn/xwiki/api/Document.java

```diff
@@ -2543,6 +2550,10 @@ public void save(String comment) throws XWikiException
     public void save(String comment, boolean minorEdit) throws XWikiException
```

Comment:
That's only one entry point to save the document and from what I understand it's not set at all in others, which I don't think is what we want.

Reply:
I might miss something there, but from what I recall, you should have the rights to set the authors when using `saveAsAuthor`, which you don't have when you use this API.

Reply:
My point is that from what I understand if you use #saveAsAuthor the display author is simply not set at all which mean any code which rely on it will have a wrong value.

Reply:
Fixed in saveAsAuthor.